src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -3023,28 +3114,50 @@ export class StudyViewPageStore {
 
     public getCustomChartDataCount(chartMeta: ChartMeta) {
         let uniqueKey: string = chartMeta.uniqueKey;
+        
         if (!this.customChartsPromises.hasOwnProperty(uniqueKey)) {
             this.customChartsPromises[uniqueKey] = remoteData<ClinicalDataCountWithColor[]>({
-                await: () => [this.selectedSamples],
+                await: () => {
+                    return _.includes([UniqueKey.WITH_MUTATION_DATA, UniqueKey.WITH_CNA_DATA],uniqueKey) ? [this.molecularProfileSampleCounts, this.selectedSamples] : [this.selectedSamples];
+                },
                 invoke: async () => {
-                    const result = _.reduce(this.selectedSamples.result, (acc, sample) => {
-                        const findCase = _.find(this._customChartsSelectedCases.get(uniqueKey), (selectedCase:CustomChartIdentifierWithValue) => selectedCase.sampleId === sample.sampleId);
-                        let value =  'NA';
-                        if(findCase !== undefined) {
-                            value =  findCase.value;
-                        }
+                    let result: { [id: string]: ClinicalDataCount } = {};
+                    if(_.includes([UniqueKey.WITH_MUTATION_DATA, UniqueKey.WITH_CNA_DATA],uniqueKey) && this.molecularProfileSampleCounts.result) {
 
-                        if (acc[value]) {
-                            acc[value].count = acc[value].count + 1
-                        } else {
-                            acc[value] = {
-                                value: value,
-                                count: 1
+                        let yesCount = UniqueKey.WITH_MUTATION_DATA === uniqueKey ? this.molecularProfileSampleCounts.result.numberOfMutationProfiledSamples : this.molecularProfileSampleCounts.result.numberOfCNAProfiledSamples;
+                        let noCount = UniqueKey.WITH_MUTATION_DATA === uniqueKey ? this.molecularProfileSampleCounts.result.numberOfMutationUnprofiledSamples : this.molecularProfileSampleCounts.result.numberOfCNAUnprofiledSamples;
+
+                        if(yesCount > 0) {
+                            result['YES'] = {
+                                'count': yesCount,
+                                'value': 'YES'
                             }
                         }
-                        return acc;
-                    }, {} as { [id: string]: ClinicalDataCount });
-
+                        if(noCount > 0) {
+                            result['NO'] = {
+                                'count': noCount,
+                                'value': 'NO'
+                            }
+                        }
+                    } else {
+                        result = _.reduce(this.selectedSamples.result, (acc, sample) => {
```

Comment:
i would avoid calling this the "result" because it's not actually the output of the code here.  give a more descriptive name